### ISSUE TITLE
Dashboard can't read info from PSUCache after some time.

### ISSUE THREAD

**Issue by rwidmark** [ironmansoftware/powershell-universal#1421]
### Steps to Reproduce

Sometime when my dashboards try to read out the PSUCache it get the error message: "The method or operation is not implemented. at line 76" and it continue to get this error until I restart the PSUService on the node then it start to work for a random timeframe again until I get this error again.
I have multiple PSUCaches that my dashboard reads in info from when the error occurs it happen on all of them, the only thing that solves this is to restart the PSU Service.
This issue started after upgrading to PSU 3.0.6.
My dashboard has had the same PSUCache storing since 2.x without any issues but this seem to start after 3.0.6 update.
The PSUCache are populated with scripts that runs every 2min and some scripts runs every 5min etc. but each one of them are storing the information to it's own PSUCache.
The only thing that I can somehow see is when it has more users the error happens more often, I don't have any evidence for it just something that I have seen.

The issue are the same on three different nodes.

![Screenshot 2022-07-27 at 18 19 04](https://user-images.githubusercontent.com/76907327/181302163-eb9c0218-4a6d-4e00-a559-f425ff91a7f5.png)

### Expected behavior

```console
Don't get the error message just run as it use to before.
```

### Actual behavior

```console
Get error messages when my dashboards tries to read from PSUCache that says "The method or operation is not implemented. at line 76"
```

### Environment data

- Three nodes (same issue on all)
- Windows Server 2022
- PSU 3.1.6
- PowerShell 7.2.5

**Comment by rwidmark** [comment]
@adamdriscoll I have all logs and my configs ready to be uploaded to you if you provide me with a link, I'll try to send it to you by email soon but I think it's to big.

**Comment by rwidmark** [comment]
@adamdriscoll i know it's night in the us now, but later today can you take a look at this? I have sent an mail to you with all configs etc 

This keeps happening all the time. If you need more info or logs just tell me

**Comment by adamdriscoll** [comment]
I can reproduce this in certain circumstances, and I have hunch what might be happening. 

Calling Get-PSUCache without -Integrated will work as long as you don't call Connect-PSUServer. As soon as that happens, Get-PSUCache will stop working because the HTTP version (non-integrated) is not implemented and will start throwing this exception. 

The work around is the call Get-PSUCache and Set-PSUCache with -Integrated. 

```
Get-PSUCache -Key 'SomeItem' -Integrated
```

For example, I can reproduce like this: 

```
Connect-PSUServer -ComputerName http://localhost:5000 -AppToken 'testtoken'
Get-PSUCache -Key 'SomeItem' 
```

As soon as I add -Integrated, it starts to work. 

```
Connect-PSUServer -ComputerName http://localhost:5000 -AppToken 'testtoken'
Get-PSUCache -Key 'SomeItem'  -Integrated
```

The reason this might happen sporadically is that something in the integrated environment is calling Connect-PSUServer and causing this to happen for the entire process. Restarting the service clears the value we check for that Connect-PSUServer sets and it stops working again.

**Comment by rwidmark** [comment]
> I can reproduce this in certain circumstances, and I have hunch what might be happening. 
> 
> 
> 
> Calling Get-PSUCache without -Integrated will work as long as you don't call Connect-PSUServer. As soon as that happens, Get-PSUCache will stop working because the HTTP version (non-integrated) is not implemented and will start throwing this exception. 
> 
> 
> 
> The work around is the call Get-PSUCache and Set-PSUCache with -Integrated. 
> 
> 
> 
> ```
> 
> Get-PSUCache -Key 'SomeItem' -Integrated
> 
> ```
> 
> 
> 
> For example, I can reproduce like this: 
> 
> 
> 
> ```
> 
> Connect-PSUServer -ComputerName http://localhost:5000 -AppToken 'testtoken'
> 
> Get-PSUCache -Key 'SomeItem' 
> 
> ```
> 
> 
> 
> As soon as I add -Integrated, it starts to work. 
> 
> 
> 
> ```
> 
> Connect-PSUServer -ComputerName http://localhost:5000 -AppToken 'testtoken'
> 
> Get-PSUCache -Key 'SomeItem'  -Integrated
> 
> ```
> 
> 
> 
> The reason this might happen sporadically is that something in the integrated environment is calling Connect-PSUServer and causing this to happen for the entire process. Restarting the service clears the value we check for that Connect-PSUServer sets and it stops working again. 

Thanks for the work :) I have added the switch intergreated and looking forward to the fix in 3.2.0 :)
